vspace/src/main/java/edu/asu/diging/vspace/core/data/ModuleRepository.java

```diff
@@ -3,15 +3,20 @@
 import java.util.List;
 
 import org.javers.spring.annotation.JaversSpringDataAuditable;
+import org.springframework.data.jpa.repository.Query;
 import org.springframework.data.repository.PagingAndSortingRepository;
 import org.springframework.stereotype.Repository;
 
+import edu.asu.diging.vspace.core.model.IVSpaceElement;
 import edu.asu.diging.vspace.core.model.impl.Module;
 
 @Repository
 @JaversSpringDataAuditable
 public interface ModuleRepository extends PagingAndSortingRepository<Module, String> {
 
     List<Module> findTop5ByOrderByCreationDateDesc();
+    
+    @Query("SELECT d FROM Module d WHERE d.name like %?1% OR d.description LIKE %?1%")
+    public List<IVSpaceElement> getContainingModules(String searchText);
```

Comment:
how about `findInNameOrDescription`? (you could probably also use Spring Data's automatic query generation)